docs/packaging_policy.md

```diff
@@ -1,55 +1,96 @@
 
 <!-- toc -->
 ## Contents
 
-  * [Sources](#sources)
-  * [Settings](#settings)
-  * [Options](#options)<!-- endToc -->
-
+* [Sources](#sources)
+* [Dependencies](#dependencies)
+* [Build](#build)
+* [Package](#package)
+* [Settings](#settings)
+* [Options](#options)<!-- endToc -->
 
 ## Sources
 
-**Origin of sources:** Library sources should come from an official origin like the library source code repository or the official
+**Origin of sources:**
+
+* Library sources should come from an official origin like the library source code repository or the official
 release/download webpage.
 
+* If an official source archive is available, it should be preferred over an auto-generated archive.
+
+**Source immutability:** Downloaded source code stored under `source` folder should not be modified. Any patch should be applied to the copy of this source code when a build is executed (basically in `build()` method).
+
 **Building from sources:** Recipes should always build packages from library sources.
 
 **Sources not accessible:**
 
-- Library sources that are not publicly available will not be allowed in this repository even if the license allows their redistribution.
+* Library sources that are not publicly available will not be allowed in this repository even if the license allows their redistribution.
 
-- If library sources cannot be downloaded from their official origin or cannot be consumed directly due to their
+* If library sources cannot be downloaded from their official origin or cannot be consumed directly due to their
   format, the recommendation is to contact the publisher and ask them to provide the sources in a way/format that can be consumed
   programmatically.
 
-- In case of needing those binaries to use them as a "build require" for some library, we will consider following the approach of adding it
+* In case of needing those binaries to use them as a "build require" for some library, we will consider following the approach of adding it
   as a system recipe (`<build_require>/system`) and making those binaries available in the CI machines (if the license allows it).
 
+## Dependencies
+
+* Version range is not allowed.
+
+* Specifiy explicit RREV (recipe revision) of dependencies is not allowed.
+
+* Vendoring in library source code should be removed (best effort) to avoid potential ODR violations. If upstream takes care to rename symbols, it may be acceptable.
+
+## Build
+
+* `build()` method should focus on build only, not installation. During the build, nothing should be written in `package` folder. Installation step should only occur in `package()` method.
+
+* The build itself should only rely on local files. Nothing should be downloaded from internet during this step. If external files are required, they should come from `requirements` or `build_requirements`, in addition to source files downloaded in `source()` or coming from recipe itself.
```

Comment:
Maybe allow an exception for settings dependent source code (usually arch/os dependent), where we have to download the source code at the beginning of `build()` (for example in `cspice` or `msys2` recipes).

Reply:
I wouldn't list exceptions in this documentation, it can make it unnecessary long and complex. If such a _workaround_ is needed, probably the user submitting the recipe will just propose it as the _only_ way to go..